piclassifier/piclassifier.py

```diff
@@ -0,0 +1,354 @@
+                    prediction,
+                    novelty,
+                    state,
+                ) = self.classifier.classify_frame_with_novelty(
+                    p_frame, track_prediction.state
+                )
+                track_prediction.state = state
+
+                if self.fp_index is not None:
+                    prediction[self.fp_index] *= 0.8
+                state *= 0.98
+                mass = region.mass
+                mass_weight = np.clip(mass / 20, 0.02, 1.0) ** 0.5
+                cropped_weight = 0.7 if region.was_cropped else 1.0
+
+                prediction *= mass_weight * cropped_weight
+
+                if len(track_prediction.predictions) == 0:
+                    if track_prediction.uniform_prior:
+                        smooth_prediction = np.ones([self.num_labels]) * (
+                            1 / self.num_labels
+                        )
+                    else:
+                        smooth_prediction = prediction
+                    smooth_novelty = 0.5
+                else:
+                    smooth_prediction = track_prediction.predictions[-1]
+                    smooth_novelty = track_prediction.novelties[-1]
+                    smooth_prediction = (
+                        1 - prediction_smooth
+                    ) * smooth_prediction + prediction_smooth * prediction
+                    smooth_novelty = (
+                        1 - prediction_smooth
+                    ) * smooth_novelty + prediction_smooth * novelty
+                track_prediction.classified_frame(
+                    self.clip.frame_on, smooth_prediction, smooth_novelty
+                )
+
+    def get_recent_frame(self):
+        return self.motion_detector.get_recent_frame()
+
+    def disconnected(self):
+        self.end_clip()
+        self.motion_detector.disconnected()
+
+    def skip_frame(self):
+        self.skip_classifying -= 1
+
+        if self.clip:
+            self.clip.frame_on += 1
+
+    def process_frame(self, lepton_frame):
+        start = time.time()
+        self.motion_detector.process_frame(lepton_frame)
+        if self.motion_detector.recorder.recording:
+            if self.clip is None:
+                self.new_clip()
+            self.track_extractor.process_frame(
+                self.clip, lepton_frame.pix, self.motion_detector.ffc_affected
+            )
+            if self.motion_detector.ffc_affected or self.clip.on_preview():
+                self.skip_classifying = 7
+                self.classified_consec = 0
+            elif (
+                self.motion_detector.ffc_affected is False
+                and self.clip.active_tracks
+                and self.skip_classifying <= 0
+                and not self.clip.on_preview()
+            ):
+                self.identify_last_frame()
+                self.classified_consec += 1
+                if self.classified_consec == PiClassifier.MAX_CONSEC:
+                    self.skip_classifying = 7
```

Comment:
and here? This should probably be declared as a "constant" at the top of the file.